### PR TITLE
Implement CGR for blackjack

### PR DESCRIPTION
**Pull request by ddcha3**
Implement CGR for blackjack game

### CODE REVIEW

File: cgr/cgr.py [hunk 1]
```diff
@@ -50,6 +50,19 @@ async def update_ratings_chameleon(self, ctx, players, winner):
         for player in players:
             await self.update_ratings_helper(ctx, [player], r2, 0.4, "chameleon")
 
+    async def update_ratings_blackjack(self, ctx, players, winners):
+        losers = set(players) - set(winners)
+        if (len(losers) > 0):
```
Comment (vsporeddy) [inline cgr/cgr.py:55]: you can just do `if losers:`

**Comment by TimmahC** [comment]
Where's chowdercoin

File: cgr/cgr.py [hunk 2]
```diff
@@ -50,6 +50,19 @@ async def update_ratings_chameleon(self, ctx, players, winner):
...
+            loser_rating = await self.get_average_rating(losers, "blackjack")
```
Comment (vsporeddy) [inline cgr/cgr.py:56]: `r2` should be a function of the _other team's rating_. Here you're just making it as if you're losing to your own team
`s` is a function of winning or losing, `1` for a win, `0`, for a loss, or something else for in between. Making it `0.4` every time means every game is treated like a tie

File: cgr/cgr.py [hunk 3]
```diff
@@ -50,6 +50,19 @@ async def update_ratings_chameleon(self, ctx, players, winner):
...
+            r2 = math.pow(10, loser_rating/400)
+            for loser in losers:
+                await self.update_ratings_helper(ctx, [loser], r2, 0.4, "blackjack")
```
Comment (vsporeddy) [inline cgr/cgr.py:59]: Why do you loop through if you're passing in the same params every time, you could just pass in the list losers

File: cgr/cgr_config.json [hunk 4]
```diff
@@ -13,6 +13,10 @@
         "chameleon": {
             "base_rating": 1000,
             "k_factor": 60000
+        },
+        "blackjack": {
+            "base_rating": 1000,
```
Comment (vsporeddy) [inline cgr/cgr_config.json:18]: How about for a single player game? "Chowdertron" should have a rating

**Review by TimmahC — CHANGES_REQUESTED** [review]
Doesn’t take into account where players can draw with the dealer.

File: cgr/cgr.py [hunk 5]
```diff
@@ -50,6 +50,16 @@ async def update_ratings_chameleon(self, ctx, players, winner):
...
             await self.update_ratings_helper(ctx, [player], r2, 0.4, "chameleon")
 
+    async def update_ratings_blackjack(self, ctx, players, winners):
+        losers = set(players) - set(winners)
+        # r2 represents other team's rating. For Blackjack, it will be chowdertron
+        r2 = math.pow(10, game_config["blackjack"]["ai_rating"]/400)
+        if (losers):
+            await self.update_ratings_helper(ctx, losers, r2, 0, "blackjack")
+
+        if (winners):
```
Comment (vsporeddy) [inline cgr/cgr.py:60]: Don't need parentheses around `if` conditions in Python

**Pull request merged**